apps/emqx_retainer/src/emqx_retainer_api.erl

```diff
@@ -184,7 +184,17 @@ lookup_retained(get, #{query_string := Qs}) ->
     {ok, Msgs} = emqx_retainer_mnesia:page_read(undefined, Topic, Page, Limit),
     {200, #{
         data => [format_message(Msg) || Msg <- Msgs],
-        meta => #{page => Page, limit => Limit, count => emqx_retainer_mnesia:size(?TAB_MESSAGE)}
+        meta => #{
+            page => Page,
+            limit => Limit,
+            count =>
+                case Topic of
+                    undefined ->
+                        emqx_retainer_mnesia:size(?TAB_MESSAGE);
+                    _ ->
+                        0
```

Comment:
I wonder if it makes more sense to simply not return count, or as string value `"unknown"`

Reply:
We should not break with the consistent of the data schema, and this mechanism also has be applied in other APIs